### ISSUE TITLE
[FEA]  Add a quick PR check focussing on missed doc generation against the default Spark version

### ISSUE THREAD

**Issue by gerashegalov** [NVIDIA/spark-rapids#4939]
>  forgot that buildall does not do that for us anymore, and I have to manually build with the default version to make it happen

we could change CI to push the result of `antrun:run@update_config_docs` to the PR branch instead of having it complain that the developer forgot to run it for the default version

_Originally posted by @gerashegalov in https://github.com/NVIDIA/spark-rapids/issues/4930#issuecomment-1064402605_

**Comment by pxLi** [comment]
cron/post-merge build for doc should be OK. One question might be, do we want it to be 
1. submit an auto-generated PR for doc after doc CI
2. directly force push to dev branch after doc CI (require grant admin access to CI token)

any preference?

**Comment by revans2** [comment]
So in my opinion adding more moving parts to CI is problematic.

I personally don't want my PR branch to be updated automatically. For example. I am working on a PR and push something not totally done. The branch is changed by CI without my knowledge while I am doing something else in parallel with it (remember CI takes hours to complete). Now I try to push my new branch and it fails because it is not a fast forward merge. Now I have to decide. Do I force push, not something I want to happen regularly, or do I try to merge it in and have to rebuild/test.  Also I don't think github permissions will let that happen our CI bot does not have permission to my repo.

I also don't want more moving parts in our builds. Setting up a new PR after the first one/etc is just a pain. What I want is something like RAPIDs where we can get really fast feedback on things like this, and not have to wait for hours to know that I forgot the docs.

**Comment by pxLi** [comment]
> So in my opinion adding more moving parts to CI is problematic.
> 
> I personally don't want my PR branch to be updated automatically. For example. I am working on a PR and push something not totally done. The branch is changed by CI without my knowledge while I am doing something else in parallel with it (remember CI takes hours to complete). Now I try to push my new branch and it fails because it is not a fast forward merge. Now I have to decide. Do I force push, not something I want to happen regularly, or do I try to merge it in and have to rebuild/test. Also I don't think github permissions will let that happen our CI bot does not have permission to my repo.
> 
> I also don't want more moving parts in our builds. Setting up a new PR after the first one/etc is just a pain. What I want is something like RAPIDs where we can get really fast feedback on things like this, and not have to wait for hours to know that I forgot the docs.

Thanks for the input! I think we may just need the option 1: an independent doc-ci check

So developer could simply get feedback from the independent check status like existing signoff check (auto-trigger when new commit comes)

**Comment by gerashegalov** [comment]
feedback via a dedicated doc-check is fine with me if this is the consensus. To make it fast requires putting a `verify` with skipTests  on the default Spark version as an early build step.
